### DISCUSSION TITLE
[Issue]: Could not load library libcudnn_cnn_infer.so.8.

### DISCUSSION THREAD

**Discussion by elchupacabrinski** [vladmandic/sdnext#540] [Q&A]
### Issue Description

Just did a clean install parallel to auto1111, but somehow I can't generate anything and I'm always getting the following error:
Could not load library libcudnn_cnn_infer.so.8. Error: libnvrtc.so: cannot open shared object file: No such file or directory

### Version Platform Description

Ubuntu 22.04

**Comment by vladmandic** [comment]
this is a torch issue, not an issue with the repo.
perhaps community can help - tranfering to discussions

**Comment by dirtyapenz** [comment]
I had the same issue on Ubuntu 22.04. I found the answer at https://www.tensorflow.org/install/pip

I did the following to get this installed using conda

```
conda create -n vladui python=3.10.9
conda activate vladui
conda install -c "nvidia/label/cuda-11.8.0" cuda-toolkit
python3 -m pip install nvidia-cudnn-cu11==8.7.0.84
mkdir -p $CONDA_PREFIX/etc/conda/activate.d
echo 'CUDNN_PATH=$(dirname $(python -c "import nvidia.cudnn;print(nvidia.cudnn.__file__)"))' >> $CONDA_PREFIX/etc/conda/activate.d/env_vars.sh
echo 'export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$CONDA_PREFIX/lib/:$CUDNN_PATH/lib' >> $CONDA_PREFIX/etc/conda/activate.d/env_vars.sh
source $CONDA_PREFIX/etc/conda/activate.d/env_vars.sh
cd ~
git clone https://github.com/vladmandic/automatic
cd automatic
./webui.sh
```

**Reply by MrUSBEN** [reply]
Thankyou! This worked for me on Ubuntu 22 for ComfyUI

**Reply by feceugur** [reply]
Thank you so much this worked for me on Ubuntu 20.04.6 LTS

**Reply by oyoanan** [reply]
fatal: 无法访问 'https://github.com/vladmandic/automatic/'：GnuTLS recv error (-110): TLS 链接非正常地终止了。

**Comment by permagreen-zz** [comment]
I'm on Arch and unfortunately the above solution didn't work for me. Whenever I hit generate I get `Could not load library libcudnn_cnn_infer.so.8. Error: libnvrtc.so: cannot open shared object file: No such file or directory
`

This only happens with vlad, btw, not automatic or any other webui.

**Reply by dirtyapenz** [reply]
Check that your $LD_LIBRARY_PATH is something sane.
`echo $LD_LIBRARY_PATH`

Check cuda version installed.
`nvcc --version`

vlad uses tensorflow. A1111 does not.

**Comment by Genius-pig** [comment]
I found two solutions:
https://discuss.tensorflow.org/t/wsl2-installation-failing-miserably/16236
https://stackoverflow.com/questions/76016645/tensorflow-2-12-could-not-load-library-libcudnn-cnn-infer-so-8-in-wsl2

I follow first one, it solve my problems.

**Reply by sixone-Jiang** [reply]
the first one is OK

**Comment by Bortrex** [comment]
In my case, I did not have installed nvcc driver. 
this fixed my problem:
`sudo apt install nvidia-cuda-toolkit`

**Reply by jcorbett** [reply]
this was my problem as well

**Reply by iturner72** [reply]
same :)

**Reply by ValerioSpenn** [reply]
Also for me!

**Reply by Ediblock** [reply]
Worked for me as well

**Reply by nnaaaa** [reply]
from another lib, but it still work for my problems

**Reply by hyun95roh** [reply]
You saved my life. Thank You !!!!

**Comment by rrakso** [comment]
This worked for me:

> As discussed in this [issue 516](https://github.com/microsoft/WSL/issues/8587) it is just a matter to add this in the .bashrc:
> ```bash
> export LD_LIBRARY_PATH=/usr/lib/wsl/lib:$LD_LIBRARY_PATH
> ```
> Be sure that your library is in /usr/lib/wsl/lib, to see it you can run
> ```bash
> ldconfig -p | grep cuda
> ```

_Reference: [Libcudnn_cnn_infer.so.8 library can not found](https://discuss.pytorch.org/t/libcudnn-cnn-infer-so-8-library-can-not-found/164661/2)_

**Comment by Nik-Kras** [comment]
Just a note: vast.ai machine was Ubuntu 22.04 x84_64 machine. And I tested it for this machine only.

Here is what helped me:

1. Install NVCC driver
```bash
sudo apt install nvidia-cuda-toolkit
```

2. Install cuDNN (Version 8!)

It is not trivial to install version 8!
If you go to the official website to install cuDNN and enter your machine's details, you will only get the latest version — version 9. I tried it, but it doesn't work (even when you link from calls of version 8 to version 9).

Official [Website](https://developer.nvidia.com/cudnn)

So, you would need to go to [archive](https://developer.nvidia.com/rdp/cudnn-archive) and submit your request for version 8.

After that, you will get a file called `.deb`. In my case, I got `cudnn-local-repo-ubuntu2204-8.9.7.29_1.0-1_amd64.deb`

Here is how you install it:

```bash
sudo dpkg -i cudnn-local-repo-ubuntu2204-8.9.7.29_1.0-1_amd64.deb
sudo cp /var/cudnn-local-repo-ubuntu2204-8.9.7/cudnn-local-*.gpg /usr/share/keyrings/
```

Maybe you would also need to do that:

```bash
sudo apt-get update
sudo apt-get install --reinstall libcudnn8 libcudnn8-dev libcudnn8-samples
```

3. Verify installation

```bash
ls /usr/lib/x86_64-linux-gnu/libcudnn* | grep libcudnn_ops_infer.so.8
ls /usr/lib/x86_64-linux-gnu/libcudnn* | grep libcudnn_cnn_infer.so.8
```

**Reply by rayryeng** [reply]
Thank you - this worked for me. I skipped step 1 because I am using WSL so I've already installed the CUDA toolkit using NVIDIA's recommended setup. The main problem was getting access to cuDNN v8, which I have downloaded from the archives and installed correctly.

The only other thing I had to do was make sure my `LD_LIBRARY_PATH` was pointing to where `libcudnn` was located, so I additionally did:

```bash
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/x86_64-linux-gnu/
```

Thanks again!
